conduit/pipeline/pipeline_test.go

```diff
@@ -866,3 +872,120 @@ func TestRoundOverwrite(t *testing.T) {
 		assert.Equal(t, uint64(i), pImpl.pipelineMetadata.NextRound)
 	}
 }
+
+// an importer that simply errors out when GetBlock() is called
+type errorImporter struct {
+	mock.Mock
+	genesis       *bookkeeping.Genesis
+	GetBlockCount uint64
+}
+
+var errorImporterMetadata = conduit.Metadata{
+	Name:         "error_importer",
+	Description:  "An importer that errors out whenever GetBlock() is called",
+	Deprecated:   false,
+	SampleConfig: "",
+}
+
+// New initializes an importer
+func New() importers.Importer {
+	return &errorImporter{}
+}
+
+func (e *errorImporter) Metadata() conduit.Metadata {
+	return errorImporterMetadata
+}
+
+func (e *errorImporter) Init(_ context.Context, _ plugins.PluginConfig, _ *log.Logger) (*bookkeeping.Genesis, error) {
+	return e.genesis, nil
+}
+
+func (e *errorImporter) Config() string {
+	return ""
+}
+
+func (e *errorImporter) Close() error {
+	return nil
+}
+
+func (e *errorImporter) GetBlock(_ uint64) (data.BlockData, error) {
+	e.GetBlockCount++
+	return data.BlockData{}, fmt.Errorf("error maker")
+}
+
+// TestPipelineRetryVariables tests that modifying the retry variables results in longer time taken for a pipeline to run
+func TestPipelineRetryVariables(t *testing.T) {
+	tests := []struct {
+		name          string
+		retryDelay    time.Duration
+		retryCount    uint64
+		totalDuration time.Duration
+		epsilon       time.Duration
+	}{
+		{"0 seconds", 2 * time.Second, 0, 0 * time.Second, 1 * time.Second},
+		{"2 seconds", 2 * time.Second, 1, 2 * time.Second, 1 * time.Second},
+		{"4 seconds", 2 * time.Second, 2, 4 * time.Second, 1 * time.Second},
+		{"10 seconds", 2 * time.Second, 5, 10 * time.Second, 1 * time.Second},
+	}
+	for _, testCase := range tests {
+		t.Run(testCase.name, func(t *testing.T) {
+
+			mockImporter := &errorImporter{genesis: &bookkeeping.Genesis{Network: "test"}}
+			mockImporter.On("GetBlock", mock.Anything).Return(data.BlockData{}, fmt.Errorf("error maker"))
```

Comment:
It looks like you aren't using the mock, so you can remove this and the embedded mock.Mock